### ISSUE TITLE
Precheck for http01 challenge

### ISSUE THREAD

**Issue by tjhiggins** [go-acme/lego#1394]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

We deploy traefik on behalf of customers and auto generate all of the cert resolver tags etc. By default we use http01 as the challenge type using letsencrypt. The problem we are running into is when a customer is slow in adding a CNAME to their domain provider. Lego will continuously try to issue a http01 cert via letsencrypt and eventually hit rate limits and backoff. 
```console
Error creating new order :: too many failed authorizations recently: see https://letsencrypt.org/docs/rate-limits/" providerName=letsencrypt.acme
```
This causes traefik/lego to only try and issue the cert once a day. So today when the user finally adds the CNAME they have to wait until tomorrow for lego to try again. Ideally it would never hit letsencrypt until an http request for the txt record resolves and is the correct value. This would be similar to the dns precheck which validates the dns record resolves prior to hitting letsencrypt.

Any reason this would be an issue?

**Comment by tjhiggins** [comment]
@ldez I'm happy to open a pr and add a preCheck for the http01 challenge. Any reason one doesn't already exist?

**Comment by ldez** [comment]
A precheck for HTTP01 is something weird from the point of view of the challenge and lego.
HTTP01 doesn't use any TXT records.

**Comment by tjhiggins** [comment]
Couldn't you make an http request to the challenge url and verify the challenge result? This is purely to avoid hitting rate limits where the url doesn't resolve, but the sec it does resolve we want letsencrypt to try and verify (not be rate limited). Same premise as the preCheck for dns to avoid rate limits, but a different check.

**Comment by dmke** [comment]
@ldez: If I understand @tjhiggins correctly (please correct me if I'm far off), they have a product similar to Github Pages, where customers can register a subdomain (say `foo.example.com`) and they setup traefik for them. Then customers can use their own domains (say `foo-service.io`), but for that to work, they need to setup a CNAME with their DNS provider (`foo-service.io. CNAME foo.example.com.`)

When @tjhiggins wants to reconfigure traefik, and the customer has not yet made the required DNS changes (or propagation time is large because of long TTLs), foo-service.io does not resolve to foo.example.com, and any HTTP01 challenge will fail.

The suggested feature is to simply request http://foo-service.io/.well-known/acme-challenge/token (from within Lego, before talking to the ACME provider, after starting an HTTP server), and check whether it fails. I'm not sure how traefik and Lego interact with each other, so I cannot say if it's easy to "just start an HTTP server" in the context of traefik.

**Comment by tjhiggins** [comment]
@dmke that is correct. Except that traefik already registers the well-known route so there is no need to start an Http Server. It would be requesting and seeing if we hit traefik and get the expected response. However, this should apply to all use cases outside of traefik which is why I created the issue here.

**Comment by tjhiggins** [comment]
@ldez any reason I shouldn't implement and put up a pr? I can make it disabled by default, but I'm still surprised it didn't already exist given the existence of a dns precheck.
